Formula/quick-lint-js.rb

```diff
@@ -0,0 +1,51 @@
+class QuickLintJs < Formula
+  desc "Find bugs in your JavaScript code"
+  homepage "https://quick-lint-js.com/"
+  url "https://c.quick-lint-js.com/releases/2.2.0/source/quick-lint-js-2.2.0.tar.gz"
+  sha256 "13bda2ff0c286d3e32053b93de723fda8caaa1d33849233e9502d16bb858815e"
+  license "GPL-3.0-or-later"
+  head "https://github.com/quick-lint/quick-lint-js.git", branch: "master"
+
+  depends_on "cmake" => :build
+
+  on_linux do
+    # Use Homebrew's C++ compiler in case the host's C++
+    # compiler is too old.
+    depends_on "gcc"
+  end
+
+  fails_with :clang do
+    build 1100 # Xcode 11.3.1
+    cause "Boost.JSON doesn't like Clang's std::string_view"
+  end
+
+  # quick-lint-js requires some C++17 features, thus
+  # requires GCC 8 or newer.
+  fails_with gcc: "5"
+  fails_with gcc: "6"
+  fails_with gcc: "7"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DQUICK_LINT_JS_INSTALL_EMACS_DIR=#{elisp}",
+                    "-DQUICK_LINT_JS_INSTALL_VIM_NEOVIM_TAGS=ON",
+                    *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"errors.js").write <<~EOF
+      const x = 3;
+      const x = 4;
+    EOF
+    ohai "quick-lint-js errors.js"
+    output = `quick-lint-js errors.js 2>&1`
+    puts output
+    refute_equal $CHILD_STATUS.exitstatus, 0
```

Comment:
```suggestion
    output = shell_output("#{bin}/quick-lint-js errors.js 2>&1", 1)
```
Replace `1` with whatever the exit code should be. We probably don't even need to assign the variable `output` anymore -- just use its return value directly in `assert_match`.

Reply:
> Replace `1` with whatever the exit code should be.

quick-lint-js doesn't guarantee a specific exit code on error. It only guarantees "non-0": https://quick-lint-js.com/cli/#_exit_status